### PR TITLE
webgui: create dictionary for pad/canvas/drawable classes

### PR DESCRIPTION
**Pull request by linev**
There are several side effects, especially with current TPadDrawable where std::shared_ptr<TPad> is used. Probably one should change class hierarchy here and derive TPadBase directly from TDrawable class.

Also dictionary for core classes will not be rebuild automatically - see my bug report: 
https://sft.its.cern.ch/jira/browse/ROOT-8949

For the time been - just rebuild this dictionary by hand doing:

    [shell] rm core/base/G__Core.cxx 

At the moment only empty canvas can be stored in the file.
If any sub-pad exists, ROOT will crash.

Also provide SetSize methods for the TCanvas

### CODE REVIEW

File: graf2d/gpad/v7/inc/ROOT/TPadUserCoord.hxx [hunk 1]
```diff
@@ -43,7 +43,7 @@ protected:
    TPadUserCoordBase() = default;
 
 public:
-   virtual ~TPadUserCoordBase();
+   virtual ~TPadUserCoordBase() {}
```
Comment (Axel-Naumann) [inline graf2d/gpad/v7/inc/ROOT/TPadUserCoord.hxx]: These function should really go into TPadUserCoordBase.cxx (which doesn't exist yet). Same for `~TPadLinearUserCoord()`. (Allows the linker to "pin the vtable" which otherwise gets emitted for every translation unit that includes this header...)

Can you change that in a subsequent PR?

File: graf2d/gpad/v7/inc/ROOT/TCanvas.hxx [hunk 2]
```diff
@@ -37,19 +37,13 @@ class TCanvasSharedPtrMaker;
...
 
-   /// Increment counter when canvas modified.
-   uint64_t fModified;
+   uint64_t fModified;                    ///<! Increment counter when canvas modified.
 
-   /// The painter of this canvas, bootstrapping the graphics connection.
-   /// Unmapped canvases (those that never had `Draw()` invoked) might not have
-   /// a painter.
-   std::unique_ptr<Internal::TVirtualCanvasPainter> fPainter;
+   std::unique_ptr<Internal::TVirtualCanvasPainter> fPainter;  ///<! The painter of this canvas, bootstrapping the graphics connection.
```
Comment (Axel-Naumann) [inline graf2d/gpad/v7/inc/ROOT/TCanvas.hxx]: Can we stick to pre-member comments? This makes it much nicer when writing long member comments.

Comment (linev) [inline graf2d/gpad/v7/inc/ROOT/TCanvas.hxx]: But how can I add ///<! comment?
fPainter should be excluded from I/O.
I tried it with multi-line comment and fail

File: graf2d/gpad/v7/inc/ROOT/TPadUserCoord.hxx (hunk 1 above)
Comment (linev) [inline graf2d/gpad/v7/inc/ROOT/TPadUserCoord.hxx]: Do you want separate PR or should I push new commits here?

Comment (Axel-Naumann) [inline graf2d/gpad/v7/inc/ROOT/TPadUserCoord.hxx]: Totally up to you - I didn't want to hold this up because of optimizations...

**Comment by linev** [comment]
Hi Axel, 

I  separate TPadUserCoordBase and TPadLinearUserCoord and provide source file for them.

With comments in TCanvas.hxx I have no idea how to proceed.
If I return member comments before member, I cannot disable I/O for this member.

Sergey

File: graf2d/gpad/v7/inc/ROOT/TCanvas.hxx (hunk 2 above)
Comment (Axel-Naumann) [inline graf2d/gpad/v7/inc/ROOT/TCanvas.hxx]: Indeed - I overlooked that.

Can we separate these two concerns? Talking to people this seems to be the favored new rule:
```
/// Good documentation, long, explaining what the member's value means etc.
fMember; //!
```
Objections?

Comment (linev) [inline graf2d/gpad/v7/inc/ROOT/TCanvas.hxx]: No problem - done now
